settlement/dymension/events.go

```diff
@@ -111,26 +111,35 @@ func convertToNewBatchEvent(rawEventData ctypes.ResultEvent) (*settlement.EventD
 		return nil, fmt.Errorf("missing expected attributes in event")
 	}
 
-	numBlocks, err := strconv.ParseInt(rawEventData.Events["state_update.num_blocks"][0], 10, 64)
+	numBlocks, err := strconv.ParseInt(events["state_update.num_blocks"][0], 10, 64)
 	if err != nil {
 		errs = append(errs, err)
 	}
-	startHeight, err := strconv.ParseInt(rawEventData.Events["state_update.start_height"][0], 10, 64)
+	startHeight, err := strconv.ParseInt(events["state_update.start_height"][0], 10, 64)
 	if err != nil {
 		errs = append(errs, err)
 	}
-	stateIndex, err := strconv.ParseInt(rawEventData.Events["state_update.state_info_index"][0], 10, 64)
+	stateIndex, err := strconv.ParseInt(events["state_update.state_info_index"][0], 10, 64)
 	if err != nil {
 		errs = append(errs, err)
 	}
 	if len(errs) > 0 {
 		return nil, errors.Join(errs...)
 	}
 	endHeight := uint64(startHeight + numBlocks - 1)
+
+	var nextProposer string
+	if events["state_update.next_proposer"] == nil {
+		nextProposer = ""
+	} else {
+		nextProposer = events["state_update.next_proposer"][0]
+	}
```

Comment:
what is this part used for?
if im not mistaken we only use the new batch to get informed about a new batch and than fetch the full batch from the hub for syncing by it's height (as other fields like block descriptors etc are also absent)